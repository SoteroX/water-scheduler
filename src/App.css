.App {
}
body,
html,
#root {
  height: 100%;
}

.App {
  min-height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.app-selection-container {
  margin: 0 auto;
}

.app-button-container {
  display: flex;
  justify-content: center;
}

.app-button-styles {
  cursor: pointer;
}
.active {
  background-color: #16cbf5;
  border: 1px solid #999696;
  color: white;
}

.active:focus {
  outline-color: #0d7098a3;
}
