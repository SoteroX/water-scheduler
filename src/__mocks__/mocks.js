export const mockEvents = [
  {
    id: 0,
    title: "All Day Event very long title",
    start: new Date(2015, 3, 0),
    end: new Date(2015, 3, 1)
  },
  {
    id: 1,
    title: "Long Event",
    start: new Date(2015, 3, 7),
    end: new Date(2015, 3, 10)
  }
];

export const mockEmptyEvents = [];

export const mockPlants = [
  { name: "Fiddle Leaf Fig", water_after: "7 days" },
  { name: "Croton", water_after: "7 days" },
  { name: "Spider Plant", water_after: "7 days" },
  { name: "Snake Plant", water_after: "14 days" },
  { name: "Money Tree", water_after: "14 days" },
  { name: "Dracaena", water_after: "14 days" },
  { name: "Jade", water_after: "14 days" },
  { name: "Bird's Nest Fern", water_after: "3 days" },
  { name: "Bell Pepper Plant", water_after: "3 days" },
  { name: "Strawberry Plant", water_after: "3 days" },
  { name: "Wavy Fern", water_after: "2 days" }
];

export const mockSortedPlants = {
  "7 days": [
    { name: "Fiddle Leaf Fig", water_after: "7 days" },
    { name: "Croton", water_after: "7 days" },
    { name: "Spider Plant", water_after: "7 days" }
  ],
  "14 days": [
    { name: "Snake Plant", water_after: "14 days" },
    { name: "Money Tree", water_after: "14 days" },
    { name: "Dracaena", water_after: "14 days" },
    { name: "Jade", water_after: "14 days" }
  ],
  "3 days": [
    { name: "Bird's Nest Fern", water_after: "3 days" },
    { name: "Bell Pepper Plant", water_after: "3 days" },
    { name: "Strawberry Plant", water_after: "3 days" }
  ],
  "2 days": [{ name: "Wavy Fern", water_after: "2 days" }]
};

export const mockEmptyPlants = {};

export const mockLists = [
  {
    name: "Fiddle Leaf Fig",
    water_after: "7 days",
    dates: [
      { month: 11, day: 16, year: 2019 },
      { month: 11, day: 23, year: 2019 },
      { month: 11, day: 30, year: 2019 },
      { month: 0, day: 6, year: 2020 }
    ]
  },
  {
    name: "test plant",
    water_after: "14 days",
    dates: [
      { month: 11, day: 16, year: 2019 },
      { month: 11, day: 23, year: 2019 },
      { month: 11, day: 30, year: 2019 },
      { month: 0, day: 6, year: 2020 }
    ]
  }
];
