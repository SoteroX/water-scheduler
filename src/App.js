import React, { useState } from "react";
// Import Components
import Welcome from "./components/welcome/welcome.component";
import BigCalendarContainer from "./components/big-calendar/big.calendar.container";
import ListContainer from "./components/list/list.container";
// Import data
import plants from "./data/data.json";
// Import styles
import "./App.css";

function App() {
  const [openCalendar, setOpenCalendar] = useState(true);

  const handleListButton = () => {
    if (openCalendar) {
      setOpenCalendar(false);
    }
  };

  const handleCalendarButton = () => {
    if (!openCalendar) {
      setOpenCalendar(true);
    }
  };

  return (
    <div className="App">
      <Welcome />
      <div className="app-selection-container">
        <h4>Select which VIEW you would like to use</h4>
        <div className="app-button-container">
          <button
            className={
              "app-button-styles " + (openCalendar === false ? "active" : null)
            }
            onClick={handleListButton}
          >
            List
          </button>
          <button
            className={
              "app-button-styles " + (openCalendar === true ? "active" : null)
            }
            onClick={handleCalendarButton}
          >
            Calendar
          </button>
        </div>
      </div>
      {openCalendar ? (
        <BigCalendarContainer plants={plants} />
      ) : (
        <ListContainer plants={plants} />
      )}
    </div>
  );
}

export default App;
