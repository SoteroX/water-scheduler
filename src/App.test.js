import React from "react";
import { shallow } from "enzyme";
// Import mocks
import { mockSortedPlants, mockEmptyPlants } from "./__mocks__/mocks";
// Import components
import App from "./App";
import Welcome from "./components/welcome/welcome.component";
import BigCalendarContainer from "./components/big-calendar/big.calendar.container";

describe("App component", () => {
  let wrapper;
  let welcome;
  let bigCalendar;
  let bigCalendarCrash;

  beforeEach(() => {
    welcome = <Welcome />;
    bigCalendar = <BigCalendarContainer sortedPlants={mockSortedPlants} />;
    bigCalendarCrash = <BigCalendarContainer sortedPlants={mockEmptyPlants} />;

    wrapper = shallow(<App />);
  });
  // Makes sure App component is render but Big Calendar component crashes
  it("renders App component but big calendar component crashes", () => {
    expect(wrapper.contains(welcome)).toEqual(true);
    expect(wrapper.contains(bigCalendarCrash)).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
  // Makes sure App component is render
  it("renders App component without crashing", () => {
    expect(wrapper.contains(welcome)).toEqual(true);
    expect(wrapper.contains(bigCalendar)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
});
