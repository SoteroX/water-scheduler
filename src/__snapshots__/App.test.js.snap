// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App component but big calendar component crashes 1`] = `
<div
  className="App"
>
  <Welcome />
  <div
    className="app-selection-container"
  >
    <h4>
      Select which VIEW you would like to use
    </h4>
    <div
      className="app-button-container"
    >
      <button
        className="app-button-styles null"
        onClick={[Function]}
      >
        List
      </button>
      <button
        className="app-button-styles active"
        onClick={[Function]}
      >
        Calendar
      </button>
    </div>
  </div>
  <BigCalendarContainer
    sortedPlants={
      Object {
        "14 days": Array [
          Object {
            "name": "Snake Plant",
            "water_after": "14 days",
          },
          Object {
            "name": "Money Tree",
            "water_after": "14 days",
          },
          Object {
            "name": "Dracaena",
            "water_after": "14 days",
          },
          Object {
            "name": "Jade",
            "water_after": "14 days",
          },
        ],
        "2 days": Array [
          Object {
            "name": "Wavy Fern",
            "water_after": "2 days",
          },
        ],
        "3 days": Array [
          Object {
            "name": "Bird's Nest Fern",
            "water_after": "3 days",
          },
          Object {
            "name": "Bell Pepper Plant",
            "water_after": "3 days",
          },
          Object {
            "name": "Strawberry Plant",
            "water_after": "3 days",
          },
        ],
        "7 days": Array [
          Object {
            "name": "Fiddle Leaf Fig",
            "water_after": "7 days",
          },
          Object {
            "name": "Croton",
            "water_after": "7 days",
          },
          Object {
            "name": "Spider Plant",
            "water_after": "7 days",
          },
        ],
      }
    }
  />
</div>
`;

exports[`App component renders App component without crashing 1`] = `
<div
  className="App"
>
  <Welcome />
  <div
    className="app-selection-container"
  >
    <h4>
      Select which VIEW you would like to use
    </h4>
    <div
      className="app-button-container"
    >
      <button
        className="app-button-styles null"
        onClick={[Function]}
      >
        List
      </button>
      <button
        className="app-button-styles active"
        onClick={[Function]}
      >
        Calendar
      </button>
    </div>
  </div>
  <BigCalendarContainer
    sortedPlants={
      Object {
        "14 days": Array [
          Object {
            "name": "Snake Plant",
            "water_after": "14 days",
          },
          Object {
            "name": "Money Tree",
            "water_after": "14 days",
          },
          Object {
            "name": "Dracaena",
            "water_after": "14 days",
          },
          Object {
            "name": "Jade",
            "water_after": "14 days",
          },
        ],
        "2 days": Array [
          Object {
            "name": "Wavy Fern",
            "water_after": "2 days",
          },
        ],
        "3 days": Array [
          Object {
            "name": "Bird's Nest Fern",
            "water_after": "3 days",
          },
          Object {
            "name": "Bell Pepper Plant",
            "water_after": "3 days",
          },
          Object {
            "name": "Strawberry Plant",
            "water_after": "3 days",
          },
        ],
        "7 days": Array [
          Object {
            "name": "Fiddle Leaf Fig",
            "water_after": "7 days",
          },
          Object {
            "name": "Croton",
            "water_after": "7 days",
          },
          Object {
            "name": "Spider Plant",
            "water_after": "7 days",
          },
        ],
      }
    }
  />
</div>
`;
